great_expectations/rule_based_profiler/config/base.py

```diff
@@ -408,32 +415,58 @@ def resolve_config_using_acceptable_arguments(
     ) -> "RuleBasedProfilerConfig":  # noqa: F821
         runtime_config: RuleBasedProfilerConfig = profiler.config
 
-        runtime_variables: Optional[
-            ParameterContainer
-        ] = profiler.reconcile_profiler_variables(variables)
-        runtime_variables_configs: dict = (
-            runtime_variables.to_dict()["parameter_nodes"]["variables"]["variables"]
-            or {}
+        runtime_variables: dict = profiler.reconcile_profiler_variables_as_dict(
```

Comment:
@cdkini Small worry here that this way only default reconciliation directives will be applied.  What if we want to customizer them later (that is why I am in favor of calling highest-level public methods)...

Reply:
I agree but I think our tests provide sufficient coverage to ensure we catch any breaking changes here.

I think the pros outweigh the cons but open to possibly re-evaluate down the line.